### DISCUSSION TITLE
Custom download and install UI

### DISCUSSION THREAD

**Discussion by YoavGro** [sparkle-project/Sparkle#2520] [Q&A] [answered]
Is there a way to support a custom UI for downloading and installing a new version of my app.
I'm looking in the `SPUUpdaterDelegate` and don't see any reference to the progress of the download

Thanks

**Comment by zorgiepoo** [comment, marked as answer]
You can create a SPUUpdater with a custom [SPUUserDriver](https://sparkle-project.org/documentation/api-reference/Protocols/SPUUserDriver.html) to achieve a custom updater UI. `SUPopUpTitlebarUserDriver` and `SPUCommandLineUserDriver` and `SPUStandardUserDriver` are examples in Sparkle's repository.
